megamek/src/megamek/common/weapons/ArtilleryWeaponIndirectHomingHandler.java

```diff
@@ -589,7 +589,7 @@ protected int getAMSHitsMod(Vector<Report> vPhaseReport) {
     //If you're firing missiles at a capital ship...
     @Override
     protected int calcCounterAV () {
-        if ((target == null) || !advancedPD) {
+        if (!(target instanceof Entity) || !advancedPD) {
```

Comment:
Good start. Can we investigate the possibility of firing one of these at a building that has a gun emplacement on it, and the gun emplacement has AMS? I bet it'll throw an NPE. I ask because I ran into a similar situation recently.

Reply:
Thinking about it some more, it won't throw an NPE, but it might not fire the AMS.